Classes/Hooks/DefaultUploadFolder.php

```diff
@@ -25,7 +25,7 @@ class DefaultUploadFolder
      * @param BackendUserAuthentication $backendUserAuthentication
      * @return Folder
      */
-    public function getDefaultUploadFolder($params, BackendUserAuthentication $backendUserAuthentication):Folder
```

Comment:
Please keep the return type hinting

Reply:
I would omit the return type hinting, because `$params['uploadFolder']` is not always an instance of `Folder`. For users without file mounts it is empty. In my current use case a backend user without file mounts needs to edit tt_address records. This leads to an error. Instead of return type hinting I would add a check if `$params['uploadFolder']` is an instance of `Folder` and return `$params['uploadFolder']` if it is not:

``` PHP
if (!($params['uploadFolder'] instanceof Folder)) {
    return $params['uploadFolder'];
}
```

I could create an additional pull request because this has nothing to do with reintroducing of configuration via UserTSconfig. What do you think about this?

Reply:
I readded the return type hinting. (I will create another pull request for the change I mentioned in my previous comment.)